crates/bevy_core_pipeline/src/skybox/mod.rs

```diff
@@ -0,0 +1,263 @@
+mod node;
+
+pub use node::*;
+
+use bevy_app::{App, Plugin};
+use bevy_asset::{load_internal_asset, Assets, Handle, HandleUntyped};
+use bevy_ecs::{
+    prelude::{Component, Entity},
+    query::With,
+    schedule::IntoSystemConfigs,
+    system::{Commands, Query, Res, ResMut, Resource},
+};
+use bevy_reflect::TypeUuid;
+use bevy_render::{
+    extract_component::{ExtractComponent, ExtractComponentPlugin},
+    prelude::{shape::Cube, Mesh},
+    render_asset::RenderAssets,
+    render_resource::{
+        BindGroup, BindGroupDescriptor, BindGroupEntry, BindGroupLayout, BindGroupLayoutDescriptor,
+        BindGroupLayoutEntry, BindingResource, BindingType, BlendState, BufferBindingType,
+        CachedRenderPipelineId, ColorTargetState, ColorWrites, CompareFunction, DepthBiasState,
+        DepthStencilState, FragmentState, MultisampleState, PipelineCache, PrimitiveState,
+        RenderPipelineDescriptor, SamplerBindingType, Shader, ShaderDefVal, ShaderStages,
+        ShaderType, SpecializedRenderPipeline, SpecializedRenderPipelines, StencilFaceState,
+        StencilState, TextureFormat, TextureSampleType, TextureViewDimension, VertexBufferLayout,
+        VertexState,
+    },
+    renderer::RenderDevice,
+    texture::{BevyDefault, Image},
+    view::{ExtractedView, Msaa, ViewTarget, ViewUniform, ViewUniforms},
+    Render, RenderApp, RenderSet,
+};
+
+const SKYBOX_SHADER_HANDLE: HandleUntyped =
+    HandleUntyped::weak_from_u64(Shader::TYPE_UUID, 55594763423201);
+
+pub struct SkyboxPlugin;
+
+impl Plugin for SkyboxPlugin {
+    fn build(&self, app: &mut App) {
+        load_internal_asset!(app, SKYBOX_SHADER_HANDLE, "skybox.wgsl", Shader::from_wgsl);
+
+        app.add_plugin(ExtractComponentPlugin::<Skybox>::default());
+
+        let mesh = Mesh::from(Cube::new(1.0));
+        let vertex_buffer_layout = mesh.get_mesh_vertex_buffer_layout().layout().clone();
+        let handle = app.world.resource_mut::<Assets<Mesh>>().add(mesh);
+
+        let render_app = match app.get_sub_app_mut(RenderApp) {
+            Ok(render_app) => render_app,
+            Err(_) => return,
+        };
+
+        let render_device = render_app.world.resource::<RenderDevice>().clone();
+
+        render_app
+            .insert_resource(SkyboxMesh { handle })
+            .insert_resource(SkyboxPipeline::new(vertex_buffer_layout, &render_device))
+            .init_resource::<SpecializedRenderPipelines<SkyboxPipeline>>()
+            .add_systems(
+                Render,
+                (
+                    prepare_skybox_pipelines.in_set(RenderSet::Prepare),
+                    queue_skybox_bind_groups.in_set(RenderSet::Queue),
+                ),
+            );
+    }
+}
+
+/// Adds a skybox to a 3D camera, based on a cubemap texture.
+///
+/// Note that this component does not (currently) affect the scene's lighting.
+/// To do so, use `EnvironmentMapLight` alongside this component.
```

Comment:
Link to this type please.

Reply:
I can't, it's part of bevy_pbr, and this crate is bevy_core_pipeline :(